### DISCUSSION TITLE
Difference between 'styles', 'style' and 'sx'?

### DISCUSSION THREAD

**Discussion by albertbronsky** [mantinedev/mantine#680] [Q&A] [answered]
Just wanted to clarify what is the difference between these props and when should you use each one?

**Comment by rtivital** [comment, marked as answer]
`sx` add styles to root element, `styles` can be used to add styles to every nested element that has selector within component

**Reply by albertbronsky** [reply]
> `sx` add styles to root element, `styles` can be used to add styles to every nested element that has selector within component

And what about `style`?

**Reply by rtivital** [reply]
`style` works the same way as it would with react elements (`<div style={{ height: 1 }} />`), it adds inline styles to element, while `sx` and `styles` add styles via emotion (inserts in head).
